src/main/java/org/jabref/logic/integrity/AmpersandChecker.java

```diff
@@ -24,6 +26,9 @@ public List<IntegrityMessage> check(BibEntry entry) {
         List<IntegrityMessage> results = new ArrayList<>();
 
         for (Map.Entry<Field, String> field : entry.getFieldMap().entrySet()) {
```

Comment:
Maybe even cooler to integrate the filteirng directly in to the getFieldMap call? 
e.g. entry.getFieldMap().entrySet().stream().filter... 
https://mkyong.com/java8/java-8-filter-a-map-examples/

Reply:
I tried my best. Needed to use `Tuple2`  for the HtmlChecker. https://github.com/JabRef/jabref/commit/dc82d445c8bbc1ac5126947cb895e062bf6bf6b5

Reply:
I double checked that all usages do not modify the list any more